### ISSUE TITLE
Update Wubiuefi installer to the latest version 24.04 LTS (Noble Numbat)

### ISSUE THREAD

**Issue by WWarrior64** [hakuna-m/wubiuefi#347]
No problem to report, only that the installer can run the latest version of Ubuntu (24.04)

**Comment by mreymann** [comment]
I would love to have this, too! :-)

**Comment by tarunsamal** [comment]
Same here waiting for 24.04 :)

**Comment by caner-cetin** [comment]
Bump, that would be cool. 20.04 -> 22.04 -> 24.04 is not hard to do once installing 20.04 but still.

**Comment by MathiasMagnus** [comment]
@WWarrior64 Did you recompile wubiuefi or did you provide command-line options to it? How exactly did you get 24.04 working?

I tried supplying command-line args to `wubi.exe`, ones found [here](https://github.com/hakuna-m/wubiuefi/blob/1da9c1ccd10a6ec193857c953274eabdd64d5f6b/src/wubi/application.py#L243-L290), but the installer does nothing. I also tried recompiling by installing the dependencies listed in the [official docs](https://wiki.ubuntu.com/WubiGuide) (`gettext build-essential grub-pc mingw32`), but MinGW is called something else in both 22.04 and 24.04 (`mingw-w64`), but running `make` doesn't finish. 

I changed `data/isolist.ini` to have the entry

```
[Ubuntu-amd64]
max_iso_size=3700000000
arch=amd64
name=Ubuntu
packages=ubuntu-desktop
metalink=http://releases.ubuntu.com/noble/ubuntu-24.04-desktop-amd64.list
metalink2=http://cdimage.ubuntu.com/noble/daily-live/pending/noble-desktop-amd64.list
website=http://www.ubuntu.com
ordering=1
```

however upon building it errors on the step

```
grub-mkimage -O x86_64-efi -c build/winboot/wubildr-bootstrap.cfg -m build/winboot/wubildr.tar -o build/winboot/EFI/grubx64.efi \
        loadenv part_msdos part_gpt fat ntfs ext2 ntfscomp iso9660 loopback search linux linuxefi boot minicmd cat cpuid chain halt help ls reboot \
        echo test configfile gzio normal sleep memdisk tar font gfxterm gettext true efi_gop efi_uga video_bochs video_cirrus probe efifwsetup \
        all_video gfxterm_background png gfxmenu
grub-mkimage: error: cannot open `/usr/lib/grub/x86_64-efi/linuxefi.mod': No such file or directory.
```

I'm running this in WSL2, which is as close as I can get to a Linux environment without having to fire up a full-blown VM just to compile wubi.exe.

What am I doing wrong?

**Comment by WWarrior64** [comment]
> @MathiasMagnus I think there is some misunderstanding here. @WWarrior64 **_wishes_** that the installer can run the latest version of Ubuntu. I don't think he **_claims_** that the installer can run the latest version of Ubuntu!

Yes that right. Sorry, I'm not very good at English, the thing that I tried to said is that it would be good the installer can run Ubuntu 24.04 Sorry for the misinterpreting

**Comment by hakuna-m** [comment]
Sorry, for my late reply. I will create new releases for Wubiuefi soon.

**Comment by tarunsamal** [comment]
Thank you @hakuna-m

**Comment by mreymann** [comment]
Thank you! I love Wubiuefi!

**Comment by litmount** [comment]
> Sorry, for my late reply. I will create new releases for Wubiuefi soon.

Awesome! looking forward to it! thanks for your work!

**Comment by mreymann** [comment]
Will there be a new Wubi release once 24.04.1 is out?

**Comment by mreymann** [comment]
@hakuna-m Any update on this?

**Comment by MathiasMagnus** [comment]
Does anyone know if 22.04 is `dist-upgrade`-able to 24.04?

**Comment by mreymann** [comment]
Not yet, but very tempted to try.

**Comment by MathiasMagnus** [comment]
For those that wonder: it doesn't upgrade. It goes through the update but becomes unable to boot.

And FWIW, it 22.04.4 barely installs. The end of the installer drops out to init1 with I/O errors on device loop1. I suspect when Ubiquity being the forsaken dumpster fire it is tries to install GRUB onto all the connected drives, it likely fails on my ReFS second disk. Rebooting the machine during the endless error message loop brings up the system alright. Dist upgrade says that the grub-pc config is in conflict, having checked both versions, I saw no grave difference between the PC's and the package maintainer's version, I tried keeping mine or installing theirs, both resulted in borked installs.

I will be partitioning my drive, as I can't keep waiting around for longer. (This is me just writing the frustration out of me over a wasted evening, I don't expect hakuna to fix it over some random user's rants: this isn't aimed at that, we're all busy and noone's entitled to using other's people work for free.)

**Comment by raspiduino** [comment]
> For those that wonder: it doesn't upgrade. It goes through the update but becomes unable to boot.

I can confirm this. I upgraded from 22.04 to 24.04. The update succeed (I even have new GNOME apps on old Ubuntu before reboot lol) but after reboot it can no longer boot to Ubuntu.

**Comment by WinfredHuang** [comment]
@hakuna-m So it's been over half a year now. Could we have update for 24.04 and beyond?

**Comment by ero35** [comment]
> Sorry, for my late reply. I will create new releases for Wubiuefi soon.

please update...

**Comment by RealThalesGit** [comment]
its has been 2 years since the last update for 22.04, 26.04 is now releasing, hope you get motivated to do it =)

**Comment by kuwaitman** [comment]
any plans on doing the update for 26.04?
